### PR TITLE
feat(assistant builder): timeframe

### PR DESCRIPTION
**Pull request by fontanierh**
add support for custom timeframes on assistant builder

### CODE REVIEW

File: front/types/assistant/actions/retrieval.ts [hunk 1]
```diff
@@ -3,10 +3,20 @@
...
 
-export type TimeframeUnit = "hour" | "day" | "week" | "month" | "year";
+export const TIME_FRAME_UNITS = [
+  "hour",
+  "day",
+  "week",
+  "month",
+  "year",
+] as const;
+export type TimeframeUnit = (typeof TIME_FRAME_UNITS)[number];
```
Comment (philipperolet) [inline front/types/assistant/actions/retrieval.ts:17]: nit: uniformization => if it's TimeFrame then TimeFrameUnit (or change TimeFrame to Timeframe)

File: front/lib/api/assistant/actions/retrieval.ts [hunk 2]
```diff
@@ -97,6 +97,10 @@ export function timeFrameFromNow(timeFrame: TimeFrame): number {
       return now - timeFrame.duration * 30 * 24 * 60 * 60 * 1000;
     case "year":
       return now - timeFrame.duration * 365 * 24 * 60 * 60 * 1000;
+    default:
+      ((x: never) => {
+        throw new Error(`Unexpected time frame unit ${x}`);
```
Comment (philipperolet) [inline front/lib/api/assistant/actions/retrieval.ts:102]: nit: time frame => either TimeFrame or Timeframe depending (cf related nit on variable naming)

Comment (philipperolet) [reply front/lib/api/assistant/actions/retrieval.ts:102]: looks like TimeFrame is used most of the time

**Review by spolu — APPROVED** [review]

**Comment by philipperolet** [comment]
So, merged before I finished my review but LGTM :fire: modulo a mini nit

File: front/types/assistant/actions/retrieval.ts (hunk 1 above)
Comment (fontanierh) [reply front/types/assistant/actions/retrieval.ts:17]: Good point thanks! Willfix

**Pull request merged**